lib/src/cli/metrics/metrics_command.dart

```diff
@@ -0,0 +1,186 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+import 'dart:async';
+
+import 'package:args/command_runner.dart';
+
+import 'flutter_info.dart';
+import 'metrics.dart';
+import 'options.dart';
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:crypto/crypto.dart';
+import 'package:logging/logging.dart';
+import 'package:path/path.dart' as path;
+import 'package:pubspec_parse/pubspec_parse.dart';
+
+class MetricsCommand extends Command<void> {
+  @override
+  final String description =
+      'Report anonymized builder metrics to Realm';
+
+  @override
+  final String name = 'metrics';
+
+  MetricsCommand() {
+    populateOptionsParser(argParser);
+  }
+
+  @override
+  FutureOr<void>? run() async {
+    try {
+      final options = parseOptionsResult(argResults!);
+      await uploadMetrics(options);
+    } catch (e, s) {
+      _log.warning('Failed to upload metrics', e, s);
+      // We squash on runtime errors!
+      // This script is called during build (via gradle, podspec, etc.)
+      // and we don't want to be the cause of a broken build!
+    }
+  }
+}
+
+Future<void> uploadMetrics(Options options) async {
+  final pubspecPath = path.join(path.current, 'pubspec.yaml');
+  final pubspec = Pubspec.parse(await File(pubspecPath).readAsString());
+
+  hierarchicalLoggingEnabled = true;
+  _log.level = options.verbose ? Level.INFO : Level.WARNING;
+
+  if (Platform.environment['CI'] != null ||
+      Platform.environment['REALM_DISABLE_ANALYTICS'] != null) {
+    _log.info('Skipping metrics upload');
+    return;
+  }
+
+  final flutterInfo = options.flutter ? await FlutterInfo.get() : null;
```

Comment:
This will get the version of Flutter from $PATH which might be different from the currently used one. I wonder is there a way to capture that from build.gradle and from the xcode prebuild script.
A quick look shows that for Android there is this code in settings.gradle of the app `def flutterSdkPath = properties.getProperty("flutter.sdk")`
I am not sure about iOS but probably there might be a way to get this as project variable or something during build.

Reply:
After an investigation it seems we can read this file in the app root dir `.dart_tool/version`, which is changed every time a new flutter sdk version is used for the build command
```
String fileContents = new File("${prj.properties.rootDir}/../.dart_tool/version").text
println fileContents
```

I expect we can do this for iOS as well. 

Lets change our code to read this file and if this is not there in the future to fallback on getting the version from the pubspec.yaml

Reply:
Introduced --flutter-root argument that (if possible) should be set to the root of flutter sdk used during build. Also, the code now tests that the sdk constraints in the pubspec matches the info retrieved from running `flutter --version --machine` and if not, use simplified info build from `.dart_tool/version` file and the currently running dart sdk. Notice that only flutter tool updates aforementioned file, not dart.